### PR TITLE
chore(build): upgrade Operator SDK to v1.31.0

### PR DESCRIPTION
**Pull request by aali309**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits): `git commit -S -m "YOUR_COMMIT_MESSAGE"`
_______________________________________________

Fixes: #614 

## Description of the change:
This change upgrades Operator SDK v1.28.0. to the newest available version at the moment (v1.31.0.)

## Motivation for the change:
upgrade to the latest available version

## How to manually test:
1. *Insert steps here...*
2. *...*

### CODE REVIEW

File: bundle/manifests/cryostat-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -54,7 +54,7 @@ metadata:
     capabilities: Seamless Upgrades
     categories: Monitoring, Developer Tools
     containerImage: quay.io/cryostat/cryostat-operator:2.4.0-dev
-    createdAt: "2023-08-24T00:50:46Z"
+    createdAt: "2023-09-18T15:10:18Z"
```
Comment (tthvo) [inline bundle/manifests/cryostat-operator.clusterserviceversion.yaml:57]: Also need to manually update:

https://github.com/cryostatio/cryostat-operator/blob/893189eacf53652fdb85fdde6cf35f33380a3815/config/manifests/bases/cryostat-operator.clusterserviceversion.yaml#L8

Comment (tthvo) [reply bundle/manifests/cryostat-operator.clusterserviceversion.yaml:57]: If you are interested @aali309, you can handle automation for this step in issue: #494

Comment (ebaron) [reply bundle/manifests/cryostat-operator.clusterserviceversion.yaml:57]: I just tried this out: apparently we can delete the `createdAt` annotation in `config/manifests/bases/cryostat-operator.clusterserviceversion.yaml`, and Operator SDK will generate the annotation when we run `make bundle`. @aali309 Could I ask you to remove the `createdAt` annotation from this file? It should still remain in the generated CSV under `bundle/`.

Comment (tthvo) [reply bundle/manifests/cryostat-operator.clusterserviceversion.yaml:57]: Uhh niceee! That's even better!!

**Comment by ebaron** [comment]
> Looks like Makefile was updated such that `operator-sdk` can be installed and referenced from `./bin`. Tho, it seems to ignore if there is a system installation.
> 
> https://github.com/operator-framework/operator-sdk/blob/e67da35ef4fff3e471a208904b2a142b27ae32b1/testdata/go/v3/memcached-operator/Makefile#L214
> 
> We should sync with these changes?

I think this is a good idea. Let's bring in this block into our Makefile: https://github.com/operator-framework/operator-sdk/blob/8f24aa560c1b8a8ea7bbc062016347b4012e499c/testdata/go/v3/memcached-operator/Makefile#L213-L228

Then anywhere we call `operator-sdk` in the Makefile, we should replace it with `$(OPERATOR_SDK)` and add the new `operator-sdk` target as a prerequisite for the target containing the `operator-sdk` call.

Once that's done we can remove the `setup-operator-sdk` step from our GitHub workflow:
https://github.com/cryostatio/cryostat-operator/blob/17d742005fed19258088bdc060357a048a3df056/.github/workflows/test-ci-reusable.yml#L63-L65

**Comment by tthvo** [comment]
Sounds good! Make sense to me!

File: Makefile [hunk 2]
```diff
@@ -368,13 +370,30 @@ $(OPM): local-bin
...
+else
+OPERATOR_SDK = $(shell which operator-sdk)
+endif
+endif
+
 # Generate bundle manifests and metadata, then validate generated files.
 .PHONY: bundle
 bundle: manifests kustomize
-	operator-sdk generate kustomize manifests -q
+	$(OPERATOR_SDK) generate kustomize manifests -q
```
Comment (tthvo) [inline Makefile:393]: Need to add `operator-sdk` as prereq here.

File: Makefile [hunk 3]
```diff
@@ -383,7 +402,7 @@ bundle-build:
 
 .PHONY: deploy_bundle
 deploy_bundle: check_cert_manager undeploy_bundle
-	operator-sdk run bundle --install-mode $(BUNDLE_INSTALL_MODE) $(BUNDLE_IMG)
+	$(OPERATOR_SDK) run bundle --install-mode $(BUNDLE_INSTALL_MODE) $(BUNDLE_IMG)
```
Comment (tthvo) [inline Makefile:405]: And here.

File: Makefile [hunk 4]
```diff
@@ -401,7 +420,7 @@ endif
 
 .PHONY: undeploy_bundle
 undeploy_bundle:
-	- operator-sdk cleanup $(OPERATOR_NAME)
+	- $(OPERATOR_SDK) cleanup $(OPERATOR_NAME)
```
Comment (tthvo) [inline Makefile:423]: Here too.

**Review by tthvo — COMMENTED** [review]
Looks good to me!

**Comment by aali309** [comment]
ping @ebaron @andrewazores

**Review by andrewazores — APPROVED** [review]

File: bundle/manifests/cryostat-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (andrewazores) [reply bundle/manifests/cryostat-operator.clusterserviceversion.yaml:57]: @aali309 could you `make bundle` and check in the regenerated files under `bundle/`? This should reset the `createdAt` here (again).

File: Makefile [hunk 5]
```diff
@@ -161,14 +163,14 @@ $(KUSTOMIZE) build internal/images/custom-scorecard-tests/rbac/ | $(CLUSTER_CLIE
 		--docker-username="$(SCORECARD_REGISTRY_USERNAME)" --docker-password="$(SCORECARD_REGISTRY_PASSWORD)"; \
 	$(CLUSTER_CLIENT) patch sa cryostat-scorecard -n $(SCORECARD_NAMESPACE) -p '{"imagePullSecrets": [{"name": "registry-key"}]}'; \
 fi
-operator-sdk run bundle -n $(SCORECARD_NAMESPACE) --timeout 20m $(BUNDLE_IMG) --security-context-config=restricted $(SCORECARD_ARGS)
+$(OPERATOR_SDK) run bundle -n $(SCORECARD_NAMESPACE) --timeout 20m $(BUNDLE_IMG) --security-context-config=restricted $(SCORECARD_ARGS)
```
Comment (ebaron) [inline Makefile:166]: The `clean-scorecard` target could use `operator-sdk` as a prerequisite.

File: Makefile [hunk 6]
```diff
@@ -368,22 +370,39 @@ $(OPM): local-bin
...
 
 # Build the bundle image.
 .PHONY: bundle-build
 bundle-build:
 	$(IMAGE_BUILDER) build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
 
 .PHONY: deploy_bundle
-deploy_bundle: check_cert_manager undeploy_bundle
-	operator-sdk run bundle --install-mode $(BUNDLE_INSTALL_MODE) $(BUNDLE_IMG)
+deploy_bundle: check_cert_manager undeploy_bundle
```
Comment (ebaron) [inline Makefile:404]: This trailing space here should be deleted. Sometimes Git complains about these sorts of things.

File: Makefile [hunk 7]
```diff
@@ -13,9 +13,11 @@ BUNDLE_VERSION ?= $(IMAGE_VERSION)
 DEFAULT_NAMESPACE ?= quay.io/cryostat
 IMAGE_NAMESPACE ?= $(DEFAULT_NAMESPACE)
 OPERATOR_NAME ?= cryostat-operator
+OPERATOR_SDK_VERSION ?= v1.31.0
 CLUSTER_CLIENT ?= kubectl
 IMAGE_TAG_BASE ?= $(IMAGE_NAMESPACE)/$(OPERATOR_NAME)
 
+
```
Comment (ebaron) [inline Makefile:20]: This blank line can be deleted.

File: Makefile [hunk 8]
```diff
@@ -306,11 +307,11 @@ remove_cert_manager:
...
-       if [ $$(echo "$${api_versions}" | grep -c '^cert-manager.io/v1$$') -eq 0 ]; then if [ "$${DISABLE_SERVICE_TLS}" != "true" ]; then\
-                       echo 'cert-manager is not installed, install using "make cert_manager" or disable TLS for services by setting DISABLE_SERVICE_TLS to true' >&2\
-                       && exit 1;\
-               fi;\
-       fi
+		if [ $$(echo "$${api_versions}" | grep -c '^cert-manager.io/v1$$') -eq 0 ]; then if [ "$${DISABLE_SERVICE_TLS}" != "true" ]; then\
+						echo 'cert-manager is not installed, install using "make cert_manager" or disable TLS for services by setting DISABLE_SERVICE_TLS to true' >&2\
+					&& exit 1;\
+				fi;\
+		fi
```
Comment (ebaron) [inline Makefile:314]: Thanks for cleaning up the other trailing spaces. This one seems to have gotten a little too much indentation. Converting to tabs is fine I think, but let's do one tab == one level of indentation.

Comment (aali309) [reply Makefile:314]: sorry about that was trying to follow the format that was used/dint want to change.

**Review by ebaron — APPROVED** [review]
Looks good. Thanks @aali309!

**Comment by aali309** [comment]
> Looks good. Thanks @aali309!

No man thanks to you @ebaron @andrewazores and @tthvo

File: bundle/manifests/cryostat-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (ebaron) [reply bundle/manifests/cryostat-operator.clusterserviceversion.yaml:57]: I reran it, and the date does get updated.

**Review by ebaron — APPROVED** [review]

**Pull request merged**